src/modules/Channel/components/MessageInput/useHandleUploadFiles.tsx

```diff
@@ -91,15 +93,26 @@ export const useHandleUploadFiles = ({
       return;
     }
 
-    if (files.length === 1) {
+    // Image Compression
+    const { compressedImages, notCompressedFiles } = await compressImages({
+      files,
+      imageCompression,
+      logger,
+    });
+    const sendingFiles = compressedImages.concat(notCompressedFiles);
```

Comment:
Would this not change the image order? We should keep the order of selected files.

Reply:
What a good catch! Thanks for catching this! Let me fix it